Dockerfile

```diff
@@ -1,35 +1,51 @@
+# ---- Base Node ----
 FROM node:18.13.0-bullseye-slim as base
 
-RUN set -ex; \
-    apt update -y; \
-    apt install ca-certificates -y; \
-    update-ca-certificates
-
-FROM base as build
-
-WORKDIR /app
-
-RUN apt install make gcc g++ python3 git openssl -y;
+RUN mkdir -p /home/node/app
+RUN chown -R node:node /home/node && chmod -R 770 /home/node
+WORKDIR /home/node/app
+
+# ---- Dependencies ----
+FROM base AS dependencies
+
+COPY --chown=node:node ./package.json ./package.json prisma ./
+COPY --chown=node:node ./package-lock.json ./package-lock.json
+RUN apt-get update -y && \
+    apt-get install -y --no-install-recommends \
+      ca-certificates \
+      dumb-init \
+      make \
+      gcc \
+      g++ \
+      python3 \
+      git \
+      openssl
+USER node
+# install production dependencies
+RUN npm ci
+# separate production node_modules
+RUN cp -R node_modules prod_node_modules
+# install ALL node_modules, including 'devDependencies'
+RUN npm install
 
-COPY --chown=node:node package.json package-lock.json prisma ./
-RUN set -ex; \
-    npm install --ignore-scripts; \
-    npx prisma generate
+# ---- Build ----
+FROM dependencies as build
 
 COPY --chown=node:node . .
-RUN set -ex; \
-    npm run build; \
-    NODE_ENV=production npm prune
-
-FROM base as app
+RUN npx prisma generate && \
```

Comment:
Please use `set -ex; \` and `;\` for multi-command RUNs, it's easier to debug if something goes wrong

Reply:
When && fails, docker shows that the whole batch failed without specifics, with `set -ex` it shows a specific command which failed.